cpp/src/arrow/filesystem/path_util.h

```diff
@@ -128,6 +128,17 @@ bool IsEmptyPath(util::string_view s);
 ARROW_EXPORT
 bool IsLikelyUri(util::string_view s);
 
+class ARROW_EXPORT Globber {
+ public:
+  ~Globber();
+  explicit Globber(std::string pattern);
+  bool Matches(std::string path);
```

Comment:
There is no need to make a copy here as the parameter isn't kept anywhere after this call.
```suggestion
  bool Matches(const std::string& path);
```